test/integration/component/test_vpc.py

```diff
@@ -2490,3 +2498,23 @@ def test_21_deploy_vm_with_gateway_ip(self):
             "Failed to create VM with first ip address in the CIDR as the vm ip"
         )
         return
+
+    @attr(tags=["advanced", "intervlan"], required_hardware="false")
+    def test_22_vpn_customer_gw_with_hostname(self):
+        """
+            Test to create vpn customer gateway with hostname
+            instead of gateway ip address
+        """
+        try:
+            vpnGw = VpnCustomerGateway.create(
+                self.apiclient,
+                self.services["vpn_customer_gw"],
+                name="test_vpn_customer_gw",
+                gateway="GwWithHostName",
+                cidrlist="10.1.0.0/16"
+            )
+            self.cleanup.append(vpnGw)
+        except Exception as e:
+            self.fail("Creating vpn customer gateway with hostname\
+                      Failed with error :%s" % e)
+        return
```

Comment:
@DaanHoogland , I have added few validations based on your review comments. Can you please check and give LGTM if you are okay with those changes?